src/Features/Core/Portable/UseCollectionInitializer/ObjectCreationExpressionAnalyzer.cs

```diff
@@ -120,6 +118,11 @@ protected override void AddMatches(ArrayBuilder<TExpressionStatementSyntax> matc
             }
         }
 
+        protected override bool ShouldAnalize()
+            => _semanticModel.GetTypeInfo(_objectCreationExpression, default).Type
+                    .GetMembers(WellKnownMemberNames.CollectionInitializerAddMethodName)
```

Comment:
1. ```.Type``` may return null, so this would crash on GetMembers.
2. Do not pass default for the cancellation token.  We should have an actual cancellation token that can be passsed.

Reply:
3. This won't pick up inherited .Add methods, or extension .Add methods.